Parsimony/ParserBuilder/CommandBuilder.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+
+namespace Parsimony.ParserBuilder
+{
+    internal interface IParserBuilder { }
+
+
+    /// <summary>
+    /// A builder for commands that allows for type safety and parser logic to be encapsulated
+    /// </summary>
+    /// <typeparam name="TOption"> The type of options required to execute the command </typeparam>
+    public class CommandBuilder<TOption> : IParserBuilder
+        where TOption : notnull
+    {
+        /// <summary>
+        /// The name of the command
+        /// </summary>
+        internal string Name { get; }
+
+        /// <summary>
+        /// The help text associated with the command that will be printed if parsing fails
+        /// </summary>
+        internal string HelpText { get; } = "";
+
+        private readonly IList<IOptionBuilder<TOption>> _builders = new List<IOptionBuilder<TOption>>();
+
+        /// <summary>
+        /// Constructs a new command builder for a command with the given name
+        /// </summary>
+        //CN: TODO -- not done this yet
+        public CommandBuilder(string name)
+        {
+            if (string.IsNullOrWhiteSpace(name))
+                throw new ArgumentException("parameter cannot be null or whitespace", nameof(name));
```

Comment:
`ArgumentOutOfRangeException`?

Reply:
Read the docs on it and it doesn't seem to fit. That seems like it's supposed to handle cases where there's a set of allowable inputs and the non null value falls  outside it. 

From the docs:
>An ArgumentOutOfRangeException exception is thrown when a method is invoked and at least one of the arguments passed to the method is not null and contains an invalid value that is not a member of the set of values expected for the argument. 

Would formatexception make more sense?

Reply:
Created issue #8 to address this